physical/azure/azure.go

```diff
@@ -198,15 +211,12 @@ func (a *AzureBackend) Get(ctx context.Context, key string) (*physical.Entry, er
 func (a *AzureBackend) Delete(ctx context.Context, key string) error {
 	defer metrics.MeasureSince([]string{"azure", "delete"}, time.Now())
 
-	blob := &storage.Blob{
-		Container: a.container,
-		Name:      key,
-	}
-
 	a.permitPool.Acquire()
 	defer a.permitPool.Release()
 
-	_, err := blob.DeleteIfExists(nil)
+	blobURL := a.container.NewBlockBlobURL(key)
+	_, err := blobURL.Delete(ctx, azblob.DeleteSnapshotsOptionNone, azblob.BlobAccessConditions{})
```

Comment:
Does this return an error if the blob did not exist?

Reply:
Update: Seems like the idempotent delete test failure on `TestBackend` is due to this.

Reply:
@calvn https://github.com/Azure/azure-storage-blob-go/blob/48358e1de5110852097ebbc11c53581d64d47300/azblob/zt_examples_test.go#L940 describes this behavior. We can change it to `DeleteSnapshotsOptionInclude`, what do you think?

Reply:
I think using `DeleteSnapshotsOptionInclude ` is fine.

The failure on the test is due to the deletion returning an error if the blob does not exist, unlike the old behavior by `DeleteIfExists`. This occurs on the latest commit (`997442d`) as well.

```
    azure_test.go:98: idempotent delete: -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, /Users/calvin/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42
        ===== RESPONSE ERROR (ServiceCode=BlobNotFound) =====
        Description=The specified blob does not exist.
```

Reply:
I don't see an equivalent method to `DeleteIfExist` on the new SDK, so you might have to do some error type checking for `azblob.ServiceCodeBlobNotFound`, similar to how you've done it on `Get`.

Reply:
Sure I could catch that but the bigger question is why does the plugin tries to delete a blob that has already been deleted?

Reply:
Storage backend implementations on the delete operation tends to no-op if the data is non-existent instead of returning an error. This is true for the current implementation of the azure storage backend via the `DeleteIfExists` method. It’s technically doing a read before a delete which is slightly different than doing a delete and ignoring the error if it’s a “not found”, but the end result should be the same — we’re not returning an error if the data didn’t exist.

There is no race in the test, it’s simply trying to delete some arbitrary key value that never got written/added.

Reply:
I looked at the old implementation and it was also returning an error that was swallow by the `DeleteIfExists` convenience function. Hence, I updated the snippet to catch it and return `nil` if we hit the `ServiceCodeBlobNotFound`.